### DISCUSSION TITLE
pass-through package dependencies

### DISCUSSION THREAD

**Discussion by bl-young** [USEPA/flowsa#250] [Q&A] [answered]
Currently, flowsa has [fedelemflowlist as a dependency](https://github.com/USEPA/flowsa/blob/master/setup.py#L14) in setup.py. The practical link here is that for FBS development we regularly implement elementary flow mapping using the FEDEFL.

However, for consistency in mapping across packages, we now use a [common mapping function in esupy](https://github.com/USEPA/esupy/blob/main/esupy/mapping.py#L10). So nowhere in flowsa is fedelemflowlist imported or called. Yet as implemented, fedelemflowlist **is not** a requirement for esupy. This is because esupy is meant to serve as a supporting package leveraged in a number of places. We can imagine that requiring fedelemflowlist as a dependency of esupy could lead to a string of requirements that's not often needed in most cases.

So this workflow seems to make sense and can be handled by the [location of the import statements and catching of import errors](https://github.com/USEPA/esupy/blob/main/esupy/mapping.py#L56-L62).

The question is then is there any better way to handle this and/or to document to a user that a package like fedelemflowlist is required within flowsa as a pass through despite it never being called within the package?

**Comment by bl-young** [comment]
@a-w-beck @matthewlchambers  any thoughts here? @WesIngwersen , @catherinebirney and I were discussing the other day.

**Reply by matthewlchambers** [reply]
Two thoughts:
1. I think the current workflow is fine, and most python users advanced enough to check whether/how `fedelemflowlist` is actually used within `flowsa` are also probably comfortable with optional dependencies, since these are pretty common (e.g. `pandas` uses extra packages for several of its data importing functions, which are not necessary to just install `pandas`).
2. I'll note that the current `seea-recursive` branch does actually use `fedelemflowlist` in its `FlowByActivity.mapt_to_fedefl_list()` method.

**Comment by matthewlchambers** [comment, marked as answer]
Since there's been no further discussion, I'm going to mark this as answered.
